File: Laboratorul9/Tema2/Observable.py
from abc import ABC, abstractmethod


class Observer(ABC):
    @abstractmethod
    def update(self, message):
        pass


class Observable:
    def __init__(self):
        self.observers = []

    def attach(self, observer):
        if observer not in self.observers:
            self.observers.append(observer)

    def detach(self, observer):
        if observer in self.observers:
            self.observers.remove(observer)

    def notify_all(self, message):
        for observer in self.observers:
            observer.update(message)

File: Lab8/src/ExercitiulLaborator2/Observable.kt
package ExercitiulLaborator2

class Observable( private var observers: MutableList<Observer>) {
    public fun add(observer: Observer){
        observers.add(observer)
    }
    public fun remove(observer: Observer){
        observers.remove(observer)
    }

    public fun NotifyAll(){
        for (observer in observers)
            observer.update("")
    }

}